utils/timed_aggregator.go

```diff
@@ -0,0 +1,173 @@
+package utils
+
+import (
+	"errors"
+	"sync"
+	"time"
+)
+
+// ------------------------------------------------
+
+var (
+	errAnachronousSample = errors.New("anachronous sample")
```

Comment:
maybe export this since it's part of the interface?

Reply:
Will do.

I also think that I should do this error return on other functions like `GetAggregateAt` etc. as this cannot do negative time. Will make those changes also.